pkg/agent/manager/cache/cache.go

```diff
@@ -377,7 +386,9 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 	if trustDomainBundleChanged {
 		c.notifyAll()
 	} else {
-		c.notifyBySelectors(notifySet)
+		for _, notifySet := range notifySets {
+			c.notifyBySelectorSet(notifySet)
+		}
```

Comment:
I wonder if we now need to pass a notified set into this function so that we can not notify a single subscriber more than once? `notifyBySelectorSet` could check for `sub` membership in `notified` before notifying, adding `sub` and notifying if not.
```suggestion
		notified := allocSubscriberSet()
		for _, notifySet := range notifySets {
			c.notifyBySelectorSet(notifySet, notified)
		}
```

Reply:
Makes sense. Would it be better to do this prior to adding something to the `notifySet`?

Reply:
nevermind previous comment, I wasnt fully understanding the request.

Reply:
This change is in.